src/compiler/types.ts

```diff
@@ -2916,12 +2916,19 @@ namespace ts {
         visit(node: Node, stop: LintStopMethod, error: LintErrorMethod): void;
     }
 
-    export interface SyntacticLintProviderStatic {
-        new (typescript: typeof ts, args: any): LintWalker;
+    export interface BaseProviderStatic {
+        ["extension-kind"]: ExtensionKind;

```

Comment:
Can we use an valid identifier here, such as `kind`, instead of `"extension-kind"`?